tests/test_all_rules.py

```diff
@@ -1319,3 +1319,101 @@ def test_group_field_in_schemas(self):
                     if fld not in schema.keys():
                         self.fail(f"{self.rule_str(rule)} alert suppression field {fld} not \
                             found in ECS, Beats, or non-ecs schemas")
+
+
+class TestNewTerms(BaseRuleTest):
+    """Test new term rules."""
+
+    @unittest.skipIf(PACKAGE_STACK_VERSION < Version.parse("8.4.0"),
+                     "Test only applicable to 8.4+ stacks for new terms feature.")
+    def test_history_window_start(self):
+        """Test new terms history window start field."""
+
+        for rule in self.production_rules:
+            if rule.contents.data.type == "new_terms":
+
+                # validate history window start field exists and is correct
+                assert rule.contents.data.new_terms.history_window_start, \
+                    "new terms field found with no history_window_start field defined"
+                assert rule.contents.data.new_terms.history_window_start[0].field == "history_window_start", \
+                    f"{rule.contents.data.new_terms.history_window_start} should be 'history_window_start'"
+
+    @unittest.skipIf(PACKAGE_STACK_VERSION < Version.parse("8.4.0"),
+                     "Test only applicable to 8.4+ stacks for new terms feature.")
+    def test_new_terms_field_exists(self):
+        # validate new terms and history window start fields are correct
+        for rule in self.production_rules:
+            if rule.contents.data.type == "new_terms":
+                assert rule.contents.data.new_terms.field == "new_terms_fields", \
+                    f"{rule.contents.data.new_terms.field} should be 'new_terms_fields' for new_terms rule type"
+
+    @unittest.skipIf(PACKAGE_STACK_VERSION < Version.parse("8.4.0"),
+                     "Test only applicable to 8.4+ stacks for new terms feature.")
+    def test_new_terms_fields(self):
```

Comment:
I think this makes sense to make a unit test. Now that we have the context of all rules. We can build the schema all at once `schema.update(**integration_schemas[tag][latest_tag_compat_ver][policy_template])`, and then do just a lookup per validation check instead of building new schema objects for each rule `, _, schema = queryvalidator.get_beats_schema([], beats_version, ecs_version)`

Reply:
Agreed. Do you want to increase the scope of this issue and add this or save it for the other issue?

Reply:
I think you have most of it here already, but if you think it's too much of a lift then id say push it. We can take some time refactoring it later (although, IMO if you're not going to do it now, you might as well just go with 9812476e50d585401486a88dea414bfcd88e7280 since you plan to refactor that anyway).